### ISSUE TITLE
[Bug Report] Cache is not set to model device

### ISSUE THREAD

**Issue by loganriggs** [neelnanda-io/TransformerLens#217]
**Describe the bug**
cache is not set to the same device as the model.

**Code example**
```from transformer_lens import HookedTransformer
model = HookedTransformer.from_pretrained(model_name, device="cuda:2")
token = model.to_tokens("testing 1 2 3")
logits, cache = model.run_with_cache(token)
print(logits.get_device())
print(model.cfg.device)
print(cache["blocks.0.mlp.hook_post"].get_device())
```
Output: 
2
cuda:2
0

**System Info**
Describe the characteristic of your environment:
 * Describe how `transformer_lens` was installed (pip, docker, source, ...)
 * What OS are you using? (Linux, MacOS, Windows) 
 * Python version (We support 3.7 -3.10 currently)
pip install github version
Linux
Python: 3.10.9

**Additional context**
The model's device isn't passed into the argument here AFAIK:
https://github.com/neelnanda-io/TransformerLens/blob/main/transformer_lens/HookedTransformer.py#L370

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/neelnanda-io/TransformerLens/issues) in the repo (**required**)

**Comment by jbloomAus** [comment]
I just noticed that this would likely be fixed by a card we made recently, #246. I'll tag it there so we close it when fixed.

**Comment by jbloomAus** [comment]
solved by #250
